File: Technical_Direction/Tools/Viewer/Data/Scripts/Game/Player/player_animation_hit_state.lua
class 'CPlayerAnimationHitState' (CState)
	function CPlayerAnimationHitState:__init() 
		CState.__init(self)
	end

	function CPlayerAnimationHitState:OnEnter(_CCharacter)
		if not ( _CCharacter == nil ) then
			num = _CCharacter:get_animation_id("hit")
			_CCharacter:get_animation_model():execute_action( num, 1.0 )
		end
	end
	
	function CPlayerAnimationHitState:Execute(_CCharacter)
	end
	
	function CPlayerAnimationHitState:OnExit(_CCharacter)
	end
	
	function CPlayerAnimationHitState:OnMessage(_CCharacter, _Msg)
		return false
	end
	
	function CPlayerAnimationHitState:__Finalize()
	end

File: EngineUAB/Code/Videogame/Characters/Player/AnimationStates/PlayerAnimationHitState.cpp
#include "PlayerAnimationHitState.h"
#include "StatesMachine\Telegram.h"
#include "RenderableObjects\AnimatedModel\AnimatedInstanceModel.h"

#if defined _DEBUG
#include "Memory\MemLeaks.h"
#endif

CPlayerAnimationHitState::CPlayerAnimationHitState( CCharacter* _pCharacter, const std::string &_Name )
	: CState(_pCharacter, _Name)
{
}

CPlayerAnimationHitState::~CPlayerAnimationHitState()
{

}

void CPlayerAnimationHitState::OnEnter( CCharacter* _pCharacter )
{
	int l_iAnimID = _pCharacter->GetAnimationID("hit");
	_pCharacter->GetAnimatedModel()->ExecuteAction(l_iAnimID, 1.0f);
}

void CPlayerAnimationHitState::Execute( CCharacter* _pCharacter, float _fElapsedTime )
{

}

void CPlayerAnimationHitState::OnExit( CCharacter* _pCharacter )
{
}

bool CPlayerAnimationHitState::OnMessage( CCharacter* _pCharacter, const STelegram& _Message )
{
	return false;
}